lib/database/migrations/0005_uncategorized_default_category.dart

```diff
@@ -0,0 +1,25 @@
+// ignore_for_file: file_names
+
+import 'package:sqflite/sqflite.dart';
+import '../migration_base.dart';
+
+// Models
+import '/model/category_transaction.dart';
+
+class UncategorizedDefaultCategory extends Migration {
+  UncategorizedDefaultCategory()
+      : super(
+          version: 5,
+          description: 'Create default "Uncategorized" category',
+        );
+
+  @override
+  Future<void> up(Database db) async {
+    // Default "Uncategorized" Category
+    await db.execute('''
+      INSERT INTO `$categoryTransactionTable`(`${CategoryTransactionFields.id}`, `${CategoryTransactionFields.name}`, `${CategoryTransactionFields.type}`, `${CategoryTransactionFields.symbol}`, `${CategoryTransactionFields.color}`, `${CategoryTransactionFields.note}`, `${CategoryTransactionFields.parent}`, `${CategoryTransactionFields.deleted}`, `${CategoryTransactionFields.createdAt}`, `${CategoryTransactionFields.updatedAt}`) VALUES
+        (0, "Uncategorized", "IN", "question_mark", 0, 'This is a default category for no categorized transactions', null, '0', '${DateTime.now()}', '${DateTime.now()}'),
+        (1, "Uncategorized", "OUT", "question_mark", 0, 'This is a default category for no categorized transactions', null, '0', '${DateTime.now()}', '${DateTime.now()}');
```

Comment:
There's a problem in migration 5:
Users who don't have the demo data (i.e. they went through onboarding by adding a budget, or started from scratch and added a category afterwards) already have a categoryTransaction with ID 1, so this query fails and the app doesn't open.

There are two ways to fix it:

1. We allow those categories to have any ID (I don't remember if they must be 0 and 1), and let the auto-increment assign them.

2. Before running the query, we check if category ID 1 already exists, and if so, we assign it a different ID and update all associated transactions. Only then we insert the new categories with IDs 0 and 1.

Any other ideas?